app/packs/src/admin/UserManagement.js

```diff
@@ -256,6 +259,143 @@ export default class UserManagement extends React.Component {
     return true;
   }
 
+  handleOnDropUserFile = (data, file) => {
+    const validFileTypes = ['text/csv', 'application/vnd.ms-excel'];
+    if (!validFileTypes.includes(file.type)) { // Note that MIME type doesn't reliably indicate file type. It's only an initial guard and data is validated more thoroughly during processing.
+      this.setState({ userFileProcessingSummary: `Invalid file type ${file.type}. Please make sure to upload a CSV file.` });
+      this.newUsers = null;
+      return false;
+    }
+    this.newUsers = data;
+    for (let i = 0; i < this.newUsers.length; i++) {
+      this.newUsers[i].data.row = i + 1;
+    }
+    console.log(this.newUsers);
+  };
+
+  handleOnErrorUserFile = (err, file, inputElem, reason) => {
+    console.log(err);
+    this.newUsers = null;
+  };
+
+  handleOnRemoveUserFile = (data) => {
+    console.log(data);
+    this.newUsers = null;
+  };
+
+  handleCreateNewUsersFromFile() {
+    if (!this.validUserFile()) {
+      this.newUsers = null;
+      this.setState({ createUserMessage: 'Finished processing user file.' });
+    } else {
+      const promisedNewUsers = this.newUsers.map(user => this.createNewUserFromFile(user));
+      Promise.allSettled(promisedNewUsers)
+        .then((userResults) => {
+          this.setState({ createUserMessage: 'Finished processing user file.' });
+          this.showProcessingSummaryUserFile(userResults);
```

Comment:
showProcessingSummaryUserFile is calling setState too. basically you have 2 setState in a row with each probably going to trigger rendering. (also not highly likely but the 2nd one could finish before the 1st one )
could you refactor using one  setState ?

Reply:
See e085070.